crates/extensions/tedge_p11_server/proto/p11_grpc.proto

```diff
@@ -0,0 +1,38 @@
+// Copyright 2015 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package p11_grpc;
+
+service P11 {
+  rpc ChooseScheme (ChooseSchemeRequest) returns (ChooseSchemeResponse) {}
+  rpc Sign (SignRequest) returns (SignResponse) {}
+}
```

Comment:
The protocol will ever be hardly more complex than these two requests. There are no throughput constraints and both the client and the server are written using the same programming language. Hence, I see gRpc as adding an unnecessary weight for little benefits on this use-case.

That said, it's good to have this as a base for comparison. Can you implement a second approach (say plain json data exchanged over an unix socket, the protocol being defined using a trait and payloads serialized with serde)?